src/Microsoft.Azure.Functions.Extensions.Mcp/Trigger/McpToolTriggerBinding.cs

```diff
@@ -132,15 +133,52 @@ private Transport GetTransportInformation(CallToolExecutionContext context)
 
     public Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
     {
-        var toolProperties = GetProperties(_toolAttribute, _triggerParameter);
+        IList<IMcpToolProperty>? toolProperties = [];
+        JsonElement? inputSchema = null;
 
-        var listener = new McpToolListener(context.Executor, context.Descriptor.ShortName, _toolAttribute.ToolName, _toolAttribute.Description, toolProperties);
+        // Generate tool properties only if input schema was not generated in the worker
+        if (!_toolAttribute.UseWorkerInputSchema)
+        {
+            toolProperties = GetProperties(_toolAttribute, _triggerParameter);
+        }
+        else
+        {
+            inputSchema = GetInputSchema(_toolAttribute);
+
+            // Throw an error in case input schema was not generated in the worker extension as expected
+            if (inputSchema is null)
+            {
+                throw new McpException("Input schema was not generated in the worker extension");
+            }
+        }
+
+        var listener = new McpToolListener(context.Executor, context.Descriptor.ShortName,
+            _toolAttribute.ToolName, _toolAttribute.Description, toolProperties, inputSchema);
 
         _toolRegistry.Register(listener);
 
         return Task.FromResult<IListener>(listener);
     }
 
+    private static JsonElement? GetInputSchema(McpToolTriggerAttribute attribute)
+    {
+        if (string.IsNullOrEmpty(attribute.InputSchema))
+        {
+            return null;
+        }
+        try
+        {
+            using var doc = JsonDocument.Parse(attribute.InputSchema);
+            return doc.RootElement.Clone();
+        }
+        catch
+        {
+            // If parsing fails, return null
```

Comment:
Don't love that we're swallowing an exception here, will make it hard to figure out whats wrong. The error above "Input schema was not generated in the worker extension" could be incorrect when the worker did generate a schema but its broken for any reason.

```csharp
        catch (JsonException ex)
        {
            throw new InvalidOperationException(
                $"Failed to parse InputSchema for tool '{attribute.ToolName}'. Schema must be valid JSON.", ex);
        }
```